x/ccv/provider/keeper/relay.go

```diff
@@ -212,14 +216,15 @@ func (k Keeper) SendVSCPacketsToChain(ctx sdk.Context, chainID, channelID string
 // QueueVSCPackets queues latest validator updates for every registered consumer chain
 func (k Keeper) QueueVSCPackets(ctx sdk.Context) {
 	valUpdateID := k.GetValidatorSetUpdateId(ctx) // current valset update ID
-	// Get the validator updates from the staking module.
-	// Note: GetValidatorUpdates panics if the updates provided by the x/staking module
-	// of cosmos-sdk is invalid.
-	stakingValUpdates := k.stakingKeeper.GetValidatorUpdates(ctx)
+
+	// Get the bonded validators from the staking module.
+	bondedValidators := k.stakingKeeper.GetLastValidators(ctx)
```

Comment:
Both [`IterateLastValidatorPowers`](https://github.com/cosmos/cosmos-sdk/blob/84675a6bf1713e8458240bdcb6ae8e6503ac3711/x/staking/keeper/validator.go#L296) and [`GetLastValidators`](https://github.com/cosmos/cosmos-sdk/blob/84675a6bf1713e8458240bdcb6ae8e6503ac3711/x/staking/keeper/validator.go#L319) traverse through the same state using `LastValidatorPowerKey`. Last validators should only contain bonded validators. If this is not the case, then something is really off and `panic`-ing seems appropriate.

I see that by using `IterateLastValidatorPowers` we potentially avoid a `panic` in a case where we have more than `maxValidators` bonded validators.  We could for example, pass an anonymous function in `IterateLastValidatorPowers` that stops traversing after we've seen `maxValidators` and things would probably work okay. But it feels, even in this case, we better `panic` instead of moving forward because something is obviously wrong in the staking module.

My take is that `GetLastValidators` is simpler to use but I do not have strong feelings on this. We could move to `IterateLastValidatorPowers` as described above.